browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -7,47 +7,51 @@
 
 #include <memory>
 #include <string>
+#include <utility>
+#include <vector>
 
 #include "base/base64url.h"
 #include "base/strings/string_util.h"
 #include "brave/browser/brave_browser_process_impl.h"
 #include "brave/browser/net/url_context.h"
 #include "brave/common/network_constants.h"
-#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
-#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
 #include "brave/components/brave_shields/browser/ad_block_service.h"
 #include "brave/components/brave_shields/browser/brave_shields_util.h"
 #include "brave/components/brave_shields/browser/brave_shields_web_contents_observer.h"
 #include "brave/components/brave_shields/common/brave_shield_constants.h"
 #include "brave/grit/brave_generated_resources.h"
+#include "content/public/browser/browser_context.h"
 #include "content/public/browser/browser_thread.h"
+#include "content/public/browser/render_process_host.h"
+#include "content/public/browser/storage_partition.h"
 #include "extensions/common/url_pattern.h"
+#include "mojo/public/cpp/bindings/remote.h"
+#include "services/network/network_context.h"
 #include "ui/base/resource/resource_bundle.h"
+#include "url/url_canon.h"
 
 namespace brave {
 
-void ShouldBlockAdOnTaskRunner(std::shared_ptr<BraveRequestInfo> ctx) {
+void ShouldBlockAdOnTaskRunner(std::shared_ptr<BraveRequestInfo> ctx,
+                               base::Optional<std::string> canonical_name) {
   bool did_match_exception = false;
   std::string tab_host = ctx->tab_origin.host();
   if (!g_brave_browser_process->ad_block_service()->ShouldStartRequest(
-          ctx->request_url, ctx->resource_type, tab_host,
-          &did_match_exception, &ctx->cancel_request_explicitly,
-          &ctx->mock_data_url)) {
-    ctx->blocked_by = kAdBlocked;
-  } else if (!did_match_exception &&
-             !g_brave_browser_process->ad_block_regional_service_manager()
-                  ->ShouldStartRequest(ctx->request_url, ctx->resource_type,
-                                       tab_host, &did_match_exception,
-                                       &ctx->cancel_request_explicitly,
-                                       &ctx->mock_data_url)) {
-    ctx->blocked_by = kAdBlocked;
-  } else if (!did_match_exception &&
-             !g_brave_browser_process->ad_block_custom_filters_service()
-                  ->ShouldStartRequest(ctx->request_url, ctx->resource_type,
-                                       tab_host, &did_match_exception,
-                                       &ctx->cancel_request_explicitly,
-                                       &ctx->mock_data_url)) {
+          ctx->request_url, ctx->resource_type, tab_host, &did_match_exception,
+          &ctx->cancel_request_explicitly, &ctx->mock_data_url)) {
     ctx->blocked_by = kAdBlocked;
```

Comment:
@bridiver I've moved `ShouldStartRequest` into the main adblock service with this PR, if you'd like to take a look